selftests/functional/runner_vmimage.py

```diff
@@ -0,0 +1,211 @@
+import os
+import sys
+import unittest
+
+from avocado.core.nrunner.runnable import Runnable
+from avocado.plugins.runners.vmimage import VMImageRunner
+from avocado.utils import process
+from selftests.utils import AVOCADO, BASEDIR, TestCaseTmpDir
+
+RUNNER = f"{sys.executable} -m avocado.plugins.runners.vmimage"
+
+
+class RunnableRunVMImage(unittest.TestCase):
+    @unittest.skipUnless(
+        os.environ.get("AVOCADO_SELFTESTS_NETWORK_ENABLED", False),
+        "Network required to run these tests",
+    )
+    def test_no_kwargs(self):
+        res = process.run(f"{RUNNER} runnable-run -k vmimage", ignore_status=True)
+        self.assertIn(b"'status': 'started'", res.stdout)
+        self.assertIn(b"'status': 'finished'", res.stdout)
+        self.assertIn(b"'time': ", res.stdout)
+        self.assertEqual(res.exit_status, 0)
+
+
+class TaskRunVMImage(unittest.TestCase):
```

Comment:
IMO you don't need to test Runnable and Task separately, because from the runner point of view it is the same code base, therefore testing the vmimage runner only as runnable or task it should be enough.